proto/viam/robotreservation/v1/robotreservation.proto

```diff
@@ -0,0 +1,159 @@
+syntax = "proto3";
+
+package viam.robotreservation.v1;
+
+import "common/v1/common.proto";
+import "google/api/annotations.proto";
+import "google/protobuf/timestamp.proto";
+
+option go_package = "go.viam.com/api/robotreservation/v1";
+option java_package = "com.viam.robotreservation.v1";
+
+// RobotReservation service is used to create and manage robot reservations in a queue
+service RobotReservationService {
```

Comment:
Let's add some documentation that this is "pre-beta" and not guaranteed to be backwards compatible.